io.openems.edge.controller.ess.cycle/src/io/openems/edge/controller/ess/cycle/statemachine/StateMachine.java

```diff
@@ -50,24 +51,16 @@ public StateMachine(State initialState) {
 
 	@Override
 	public StateHandler<State, Context> getStateHandler(State state) {
-		switch (state) {
-		case START_CHARGE:
-			return new StartChargeHandler();
-		case START_DISCHARGE:
-			return new StartDischargeHandler();
-		case CONTINUE_WITH_CHARGE:
-			return new ContinueWithChargeHandler();
-		case CONTINUE_WITH_DISCHARGE:
-			return new ContinueWithDischargeHandler();
-		case COMPLETED_CYCLE:
-			return new CompletedCycleHandler();
-		case FINAL_SOC:
-			return new FinalSocHandler();
-		case FINISHED:
-			return new FinishedHandler();
-		case UNDEFINED:
-			return new UndefinedHandler();
-		}
-		throw new IllegalArgumentException("Unknown State [" + state + "]");
+		return switch (state) {
+		case UNDEFINED -> new UndefinedHandler();
```

Comment:
just a general question why isnt the factory of the handler included in the enum?